client/my-sites/people/delete-user/index.jsx

```diff
@@ -308,17 +310,88 @@ class DeleteUser extends React.Component {
 
 	render() {
 		// A user should not be able to remove themself.
-		if ( ! this.props.isJetpack && this.props.user.ID === this.props.currentUser.ID ) {
-			return null;
-		}
-		if ( this.props.isJetpack && this.props.user.linked_user_ID === this.props.currentUser.ID ) {
+		const { user, currentUser } = this.props;
+		const isCurrentUser = ( user.linked_user_ID ?? user.ID ) === currentUser.ID;
+
+		if ( isCurrentUser ) {
 			return null;
 		}
 
 		return this.props.isMultisite ? this.renderMultisite() : this.renderSingleSite();
 	}
 }
 
+const withDeleteUser = ( Component ) => {
+	return ( props ) => {
+		const { siteId, user, translate, isMultisite, siteSlug } = props;
+		const { deleteUser, isSuccess, isError, error } = useDeleteUser( siteId );
+		const dispatch = useDispatch();
+
+		React.useEffect( () => {
+			if ( isSuccess ) {
+				dispatch(
+					successNotice(
+						isMultisite
+							? translate( 'Successfully removed @%(user)s', {
+									args: { user: user?.login },
+									context: 'Success message after a user has been removed.',
+							  } )
+							: translate( 'Successfully deleted @%(user)s', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'Successfully deleted @%(user)s', { context: 'Success message after a user has been modified.'} )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/try/users-react-query-experiment-edit-team)


:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?